apis/project.cattle.io/v3/schema/schema.go

```diff
@@ -1020,3 +1022,70 @@ func monitoringTypes(schemas *types.Schemas) *types.Schemas {
 		).
 		MustImport(&Version, monitoringv1.Alertmanager{}, projectOverride{})
 }
+
+func autoscalingTypes(schemas *types.Schemas) *types.Schemas {
+	return schemas.
+		AddMapperForType(&Version, autoscaling.HorizontalPodAutoscaler{},
+			&m.ChangeType{Field: "scaleTargetRef", Type: "reference[workload]"},
+			&m.Move{From: "scaleTargetRef", To: "workloadId"},
+			mapper.CrossVersionObjectToWorkload{Field: "workloadId"},
+			&m.Required{Fields: []string{"workloadId", "maxReplicas"}},
+			&m.AnnotationField{Field: "displayName"},
+			&m.DisplayName{},
+			&m.AnnotationField{Field: "description"},
+			&m.Embed{Field: "status"},
+			mapper.NewMergeListByIndexMapper("currentMetrics", "metrics", "type"),
+		).
+		AddMapperForType(&Version, autoscaling.MetricTarget{},
+			&mapper.IntOrStringParser{Field: "averageUtilization"},
+			&m.Enum{Field: "type", Options: []string{"Utilization", "Value", "AverageValue"}},
+			&m.Move{To: "utilization", From: "averageUtilization"},
```

Comment:
Why do we want to trim the `average` in `averageUtilization`?

Reply:
Because we have options in type field and I want to keep the consistency
- type `Value` -> field `value`
- type `AverageValue` -> field `averageValue`
- type `Utilization` -> field `utilization`